RTClib.cpp

```diff
@@ -118,6 +118,45 @@ static void write_i2c_register(uint8_t addr, uint8_t reg, uint8_t val) {
 const uint8_t daysInMonth[] PROGMEM = {31, 28, 31, 30, 31, 30,
                                        31, 31, 30, 31, 30};
 
+/**************************************************************************/
+/*!
+    @brief checks if the year is a leap year
+    @param year The year to checks
+    @return true if a leap year, false otherwise
+*/
+/**************************************************************************/
+bool isLeapYear(uint16_t year) {
+  return year % 400 == 0 || (year % 4 == 0 && year % 100 != 0);
```

Comment:
Within the range of supported years (2000–2099), `year % 4 == 0` is enough. And this test is already used in a [couple](../../blob/1.11.0/RTClib.cpp#L137) of [places](../../blob/1.11.0/RTClib.cpp#L196) in the current code.

Reply:
Simplified `isLeapYear()` and made it static to prevent misuse.